### PR TITLE
various fixes for 0.7

### PR DESCRIPTION
**Pull request by ExpandingMan**
These are a bunch of simple fixes for 0.7.  There are still some outstanding problems.  In particular, wrong variables seem to get substituted in some cases.  It's almost as if DataFramesMeta is depending on ordering of `Expr` args somewhere where it shouldn't be, but it wasn't easy to find the problem (and I didn't).  This is a good first step at least.

### CODE REVIEW

File: src/compositedataframe.jl [hunk 1]
```diff
@@ -138,7 +138,10 @@ DataFrames.index(cdf::AbstractCompositeDataFrame) = DataFrames.Index(names(cdf))
 #########################################
 
 Base.getindex(cdf::AbstractCompositeDataFrame, col_inds::DataFrames.ColumnIndex) = getfield(cdf, col_inds)
-Base.getindex{T <: DataFrames.ColumnIndex}(cdf::AbstractCompositeDataFrame, col_inds::AbstractVector{T}) = CompositeDataFrame(Any[ getfield(cdf, col_inds[i]) for i = 1:length(col_inds) ], names(cdf)[col_inds])
+function Base.getindex(cdf::AbstractCompositeDataFrame, col_inds::AbstractVector{T}
+                      ) where {T<:DataFrames.ColumnIndex}
```
Comment (nalimilan) [inline src/compositedataframe.jl]: I'd keep this on the previous line, I don't think it's common to start a line with `)`.

File: test/dataframes.jl [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 using DataFramesMeta
 
 df = DataFrame(A = 1:3, B = [2, 1, 2])
 
 x = [2, 1, 0]
 
-@test  @with(df, :A + 1)   ==  df[:A] + 1
-@test  @with(df, :A + :B)  ==  df[:A] + df[:B]
-@test  @with(df, :A + x)   ==  df[:A] + x
+@test  @with(df, :A .+ 1)   ==  df[:A] + 1
```
Comment (nalimilan) [inline test/dataframes.jl]: Shouldn't `.+` be used on the RHS too?

**Comment by tshort** [comment]
I'm sorry for letting this slip. Is this okay to merge now? I see it still passes on Travis on v0.6.

**Comment by nalimilan** [comment]
Should still be OK, if conflicts can be fixed.

**Pull request merged**